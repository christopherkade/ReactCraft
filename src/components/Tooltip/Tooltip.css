.rc-tooltip {
  display: inline-block;
  position: relative;
  cursor: default;
  white-space: nowrap;
}

.rc-tooltip-text {
  background-color: rgb(47, 27, 37, 0.8);
  color: white;
  -webkit-text-stroke-width: 0.3px;
  -webkit-text-stroke-color: black;
  text-shadow: 1.2px 1.5px black;
  border: 1px solid rgb(209, 207, 208);
  border-radius: 3px;
  padding: 0.2rem 0.5rem;
  position: absolute;
  top: -30px;
  right: -50px;
  text-align: center;
  z-index: 100;
  box-shadow: inset 1px 1px 1rem 0rem black;
}
