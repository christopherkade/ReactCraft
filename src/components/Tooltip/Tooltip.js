import React, { useState } from "react"
import PropTypes from "prop-types"

import "./Tooltip.css"

const Tooltip = ({ children, delay, text }) => {
  const [show, setShow] = useState(false);
  const [delayTimer, setDelayTimer] = useState(null);

  const handleEnter = () => {
    const timer = setTimeout(() => {
      setShow(true);
    }, delay);
    setDelayTimer(timer);
  };

  const handleLeave = () => {
    clearTimeout(delayTimer);
    setShow(false);
  };

  return (
    <div className="rc-tooltip" onMouseEnter={handleEnter} onMouseLeave={handleLeave}>
      {show ? <div class="rc-tooltip-text">{text}</div> : null}
      {children}
    </div >
  )
}

Tooltip.propTypes = {
  children: PropTypes.node.isRequired,
  text: PropTypes.string,
  delay: PropTypes.number,
};

Tooltip.defaultProps = {
  delay: 150,
  text: "Tooltip",
};

export default Tooltip
