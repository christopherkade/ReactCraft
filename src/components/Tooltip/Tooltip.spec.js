import React from 'react';
import Tooltip from './Tooltip';
import { render } from 'react-testing-library';

describe('<Tooltip />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Tooltip text="Hovered">Tooltip text</Tooltip>);
      expect(container).toMatchSnapshot();
    });
  });

  describe('tile prop', () => {
    it('should display Hover me! as the Tooltip\'s child', () => {
      const { container } = render(<Tooltip text="Hovered">Hover me!</Tooltip>);
      expect(container.querySelector('div').textContent).toBe('Hover me!');
    });
  });
});
