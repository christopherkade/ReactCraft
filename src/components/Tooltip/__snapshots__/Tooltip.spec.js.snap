// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="rc-tooltip"
  >
    Tooltip text
  </div>
</div>
`;
